File: go-cli/commands.go
package main

import "strings"

const helpText = `Simple CLI commands:
  /help                 Show this message
  /exit                 Quit the CLI
  /model <name>         Select a model
  /provider <name>      Select a provider
  /agent <name>         Select an agent`

type commandResult struct {
    Type string
    Key  string
    Val  string
}

func parseCommand(input string) commandResult {
    parts := strings.Fields(strings.TrimPrefix(strings.TrimSpace(input), "/"))
    if len(parts) == 0 {
        return commandResult{Type: "unknown"}
    }
    switch parts[0] {
    case "exit", "quit":
        return commandResult{Type: "exit"}
    case "help":
        return commandResult{Type: "help"}
    case "model":
        return commandResult{Type: "set", Key: "model", Val: strings.Join(parts[1:], " ")}
    case "provider":
        return commandResult{Type: "set", Key: "provider", Val: strings.Join(parts[1:], " ")}
    case "agent":
        return commandResult{Type: "set", Key: "agent", Val: strings.Join(parts[1:], " ")}
    default:
        return commandResult{Type: "unknown", Val: input}
    }
}

func applyCommand(cfg *ResolvedConfig, cmd commandResult) {
    if cmd.Type != "set" {
        return
    }
    switch cmd.Key {
    case "model":
        cfg.Model = cmd.Val
    case "provider":
        cfg.Provider = cmd.Val
    case "agent":
        cfg.Agent = cmd.Val
    }
}

File: packages/simple-cli/src/commands.ts
import type { ResolvedConfig } from "./types"

export type CommandResult =
  | { type: "exit" }
  | { type: "help" }
  | { type: "set"; key: "model" | "agent" | "provider"; value?: string }
  | { type: "unknown"; input: string }

export function parseCommand(input: string): CommandResult {
  const [command, ...rest] = input.slice(1).trim().split(/\s+/)
  const value = rest.join(" ") || undefined
  switch (command) {
    case "exit":
    case "quit":
      return { type: "exit" }
    case "help":
      return { type: "help" }
    case "model":
      return { type: "set", key: "model", value }
    case "agent":
      return { type: "set", key: "agent", value }
    case "provider":
      return { type: "set", key: "provider", value }
    default:
      return { type: "unknown", input: input.trim() }
  }
}

export function helpText(): string {
  return `Simple CLI commands:\n` +
    `  /help                 Show this message\n` +
    `  /exit                 Quit the CLI\n` +
    `  /model <name>         Select a model\n` +
    `  /provider <name>      Select a provider\n` +
    `  /agent <name>         Select an agent\n`
}

export function applyCommand(result: CommandResult, config: ResolvedConfig): ResolvedConfig {
  if (result.type !== "set") return config
  return { ...config, [result.key]: result.value }
}
